module/Frontpage/src/Frontpage/Service/Poll.php

```diff
@@ -25,10 +25,19 @@ public function getNewestPoll()
      * Retrieves a poll by its id
      * @param int $pollId the id of the poll to retrieve
      * @return PollModel|null
+     *
+     * @throws \User\Permissions\NotAllowedException if the user isn't allowed to see unapproved polls
      */
     public function getPoll($pollId)
     {
-        return $this->getPollMapper()->findPollById($pollId);
+        $poll = $this->getPollMapper()->findPollById($pollId);
+        if ((null === $poll->getApprover()) && !$this->isAllowed('view_unapproved')) {
```

Comment:
Please refrain from Yoda conditions.

Also === null breaks consistency with the is_null that is used in the rest of the code. is_null has another benefit, that it does not allow you to accidentally type == null